demos/jobs_for_aws/jobs_demo.c

```diff
@@ -0,0 +1,789 @@
+            {
+                LogWarn( ( "Received an unexpected messages from AWS IoT Jobs service: "
+                           "JobsTopicType=%u", topicType ) );
+            }
+        }
+        else if( xStatus == JobsNoMatch )
+        {
+            LogWarn( ( "Incoming message topic does not belong to IoT Jobs!: topic=%.*s",
+                       pxDeserializedInfo->pPublishInfo->topicNameLength,
+                       ( const char * ) pxDeserializedInfo->pPublishInfo->pTopicName ) );
+        }
+        else
+        {
+            LogError( ( "Failed to parse incoming publish job. Topic=%.*s!",
+                        pxDeserializedInfo->pPublishInfo->topicNameLength,
+                        ( const char * ) pxDeserializedInfo->pPublishInfo->pTopicName ) );
+        }
+    }
+    else
+    {
+        vHandleOtherIncomingPacket( pxPacketInfo, usPacketIdentifier );
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Entry point of the Jobs demo.
+ *
+ * This main function demonstrates how to use the Jobs library API Device Shadow library.
+ *
+ * The helper functions this demo uses for MQTT operations have internal
+ * loops to process incoming messages. Those are not the focus of this demo
+ * and therefor, are placed in a separate file mqtt_demo_utils.c.
+ */
+int RunJobsDemo( bool awsIotMqttMode,
+                 const char * pIdentifier,
+                 void * pNetworkServerInfo,
+                 void * pNetworkCredentialInfo,
+                 const void * pNetworkInterface )
+{
+    BaseType_t xDemoStatus = pdPASS;
+
+    /* Remove compiler warnings about unused parameters. */
+    ( void ) awsIotMqttMode;
+    ( void ) pIdentifier;
+    ( void ) pNetworkServerInfo;
+    ( void ) pNetworkCredentialInfo;
+    ( void ) pNetworkInterface;
+
+    /* Establish an MQTT connection with AWS IoT over a mutually authenticated TLS session */
+    xDemoStatus = EstablishMqttSession( &xMqttContext,
+                                        &xNetworkContext,
+                                        &xBuffer,
+                                        prvEventCallback );
+
+    if( xDemoStatus == pdFAIL )
+    {
+        /* Log error to indicate connection failure. */
+        LogError( ( "Failed to connect to AWS IoT broker." ) );
+    }
+    else
+    {
+        /* Print out a short user guide to the console. The default logging
+         * limit of 255 characters can be changed in demo_logging.c, but breaking
+         * up the only instance of a 1000+ character string is more practical. */
+        LogInfo( ( "\r\n"
+                   "/*-----------------------------------------------------------*/\r\n"
+                   "\r\n"
+                   "The Jobs demo is now ready to accept Jobs.\r\n"
+                   "Jobs may be created using the AWS IoT console or AWS CLI.\r\n"
+                   "See the following link for more information.\r\n"
+                   "\r\n" ) );
```

Comment:
Do we want the extra `\r\n` at the end of the LogInfo?

Reply:
Good point. Don't need the extra line break